src/item_location.cpp

```diff
@@ -31,7 +33,12 @@ class item_location::impl
         virtual void remove_item() = 0;
 
     protected:
-        item *what = nullptr;
+        impl( item_uid &&uid ) : uid( std::move( uid ) ) {}
```

Comment:
The other option is implementing custom copy constructors for `item` and any other classes that later gain serializable id's which is less preferable.